tests/tracking/test_mlflow_artifacts.py

```diff
@@ -321,3 +326,41 @@ def test_rest_get_model_version_artifact_api_proxied_artifact_root(artifacts_ser
     )
     get_model_version_artifact_response.raise_for_status()
     assert get_model_version_artifact_response.text == "abcdefg"
+
+
+@pytest.mark.parametrize(
+    ("filename", "expected_mime_type"),
+    [
+        ("a.txt", "text/plain"),
+        ("b.pkl", "application/octet-stream"),
+        ("c.png", "image/png"),
+        ("d.pdf", "application/pdf"),
+        ("MLmodel", "text/plain"),
+        ("mlproject", "text/plain"),
+    ],
+)
+def test_mime_type_for_download_artifacts_api(
+    artifacts_server, tmpdir, filename, expected_mime_type
+):
+    default_artifact_root = artifacts_server.default_artifact_root
+    url = artifacts_server.url
+    test_file = tmpdir.join(filename)
+    test_file.write(None)
+    upload_file(test_file, f"{default_artifact_root}/dir/{filename}")
+    download_response = download_file(f"{default_artifact_root}/dir/{filename}", test_file)
+
+    _, params = cgi.parse_header(download_response.headers["Content-Disposition"])
+    assert params["filename"] == filename
+    assert download_response.headers["Content-Type"] == expected_mime_type
+
+    mlflow.set_tracking_uri(url)
+    with mlflow.start_run() as run:
+        mlflow.log_artifact(test_file)
+    artifact_response = requests.get(
+        url=f"{url}/get-artifact", params={"run_id": run.info.run_id, "path": filename}
+    )
+    artifact_response.raise_for_status()
+    _, params = cgi.parse_header(artifact_response.headers["Content-Disposition"])
+    assert params["filename"] == filename
+    assert artifact_response.headers["Content-Type"] == expected_mime_type
+    assert artifact_response.headers["X-Content-Type-Options"] == "nosniff"
```

Comment:
Can we add a test where we write a particular file with an extension, but in the request we explicitly add a different mimetype header to simulate nefarious behavior and ensure that the server returns the correct mimetype and ignores the requested type?